### PR TITLE
find debug libraries correctly

### PR DESCRIPTION
**Pull request by Karsten1987**
fixes the windows debug compilation

debug windows: [Build Status](http://ci.ros2.org/job/ci_windows/2461/)
release windwos: [Build Status](http://ci.ros2.org/job/ci_windows/2462/)

### CODE REVIEW

**Comment by Karsten1987** [comment]
I failed configuring our buildfarm jobs correctly, so here's a new set of builds:

debug: [Build Status](http://ci.ros2.org/job/ci_windows/2466/)
release: [Build Status](http://ci.ros2.org/job/ci_windows/2465/)

**Comment by Karsten1987** [comment]
compiles 🎉 
warnings are unrelated.

File: cmake/urdfdom-config.cmake.in [hunk 1]
```diff
@@ -7,16 +7,31 @@ set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include" "@TinyXML_INCLUDE_D
...
+  if (onelib)
+    list(APPEND @PKG_NAME@_RELEASE_LIBRARIES ${onelib})
+  endif()
+  if (onelibd)
+    list(APPEND @PKG_NAME@_DEBUG_LIBRARIES ${onelibd})
+  endif()
 endforeach()
-
+if(@PKG_NAME@_RELEASE_LIBRARIES)
+  list(APPEND @PKG_NAME@_LIBRARIES optimized ${@PKG_NAME@_RELEASE_LIBRARIES})
```
Comment (dirk-thomas) [inline cmake/urdfdom-config.cmake.in]: The build configuration keyword must be stated before each library name (not only once in front of multiple names).

I would suggest to fill `@PKG_NAME@_LIBRARIES` directly in line 23 and 26 and avoid the temporary variables `@PKG_NAME@_RELEASE_LIBRARIES` and `@PKG_NAME@_DEBUG_LIBRARIES`.

**Comment by Karsten1987** [comment]
new round of builds:

DEbug: [Build Status](http://ci.ros2.org/job/ci_windows/2472/)
release: [Build Status](http://ci.ros2.org/job/ci_windows/2473/)

**Comment by Karsten1987** [comment]
🎉 🌮
Compiles with no warnings from this package.

@dirk-thomas can you verify this?

File: cmake/urdfdom-config.cmake.in [hunk 2]
```diff
@@ -7,17 +7,26 @@ set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include" "@TinyXML_INCLUDE_D
...
   set(onelib "${lib}-NOTFOUND")
+  set(onelibd "${lib}-NOTFOUND")
   find_library(onelib ${lib}
       PATHS "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@"
     NO_DEFAULT_PATH
     )
-  if(NOT onelib)
+  find_library(onelibd ${lib}@CMAKE_DEBUG_POSTFIX@
+      PATHS "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@"
+    NO_DEFAULT_PATH
```
Comment (dirk-thomas) [inline cmake/urdfdom-config.cmake.in]: The indentation seems to be off (same as in the untouched `find_library` call above).

File: cmake/urdfdom-config.cmake.in [hunk 3]
```diff
@@ -7,17 +7,26 @@ set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include" "@TinyXML_INCLUDE_D
...
+    )
+  if(onelib-NOTFOUND AND onelibd-NOTFOUND)
     message(FATAL_ERROR "Library '${lib}' in package @PKG_NAME@ is not installed properly")
   endif()
-  list(APPEND @PKG_NAME@_LIBRARIES ${onelib})
+  if (onelib)
```
Comment (dirk-thomas) [inline cmake/urdfdom-config.cmake.in]: Nitpick: no space between `if` and `(`.

Same below.

**Review by dirk-thomas — COMMENTED** [review]
It looks fine to me but I haven't tried to compile / use it.

**Comment by Karsten1987** [comment]
updated style.

**Review by mikaelarguedas — APPROVED** [review]
:+1: to merge this if it gets the nightlies to build again. We can iterate on this if it proves not to be working as expected

**Comment by wjwwood** [comment]
I'm going to merge this following @mikaelarguedas' recommendation.

**Comment by wjwwood** [comment]
Does this address only the Windows failures? Or does it aim to fix the failures on the OS X nightly and packaging job too? Those look like compiler errors.

**Comment by mikaelarguedas** [comment]
I think that this addressed primarily the windows failures. Not sure about the osx ones because it seemed that it built fine in release and debug but failed on None and RelWithDebinfo...

**Comment by Karsten1987** [comment]
Is there anything else than "optimized" or "debug" in terms of `RelWithDebInfo` ?

**Comment by mikaelarguedas** [comment]
Not that I know of, there is a "general" but my understanding is that it is
for both release and debug.

Did you have to install eigen on the mac machines to get your builds
working ? It looks lie the same build passed on mini1, and dosa failed 
with a "None" build type, so it looks like a dependency issue on dosa 
more than a cmake problem IMO.

**Pull request merged**